### PR TITLE
Stack tracing refactoring

### PR DESCRIPTION
**Pull request by clover2123**
* remove redundant structures in stack tracing
* rename StackTrace structures

Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

File: src/api/EscargotPublic.cpp [hunk 1]
```diff
@@ -1194,12 +1194,12 @@ void DebuggerOperationsRef::BreakpointOperations::getStackTrace(DebuggerOperatio
     outStackTrace.clear();
 
     for (uint32_t i = 0; i < size; i++) {
-        if ((size_t)stackTraceDataVector[i].second.loc.actualCodeBlock != SIZE_MAX) {
-            ByteCodeBlock* byteCodeBlock = stackTraceDataVector[i].second.loc.actualCodeBlock;
+        if ((size_t)stackTraceDataVector[i].loc.actualCodeBlock != SIZE_MAX) {
```
Comment (bbrto21) [inline src/api/EscargotPublic.cpp:1197]: if you don't mind, could you replace c-style casting with c++ style casting? 
(It is not required, so you can ignore it :smiley:)

File: src/runtime/SandBox.cpp [hunk 2]
```diff
@@ -174,8 +166,8 @@ bool SandBox::createStackTrace(StackTraceDataVector& stackTraceDataVector, Execu
 
         bool alreadyExists = false;
 
-        for (size_t i = 0; i < stackTraceDataVector.size(); i++) {
-            if (stackTraceDataVector[i].first == es || stackTraceDataVector[i].first->lexicalEnvironment() == es->lexicalEnvironment()) {
+        for (size_t i = 0; i < stateStack.size(); i++) {
+            if (stateStack[i] == es || stateStack[i]->lexicalEnvironment() == es->lexicalEnvironment()) {
```
Comment (bbrto21) [inline src/runtime/SandBox.cpp:170]: The `stateStack` is created at line 143.
But until it get here, I couldn't find place where adding items to stateStack. 
I think I'm missing something, can you explain?

File: src/api/EscargotPublic.cpp (hunk 1 above)
Comment (clover2123) [reply src/api/EscargotPublic.cpp:1197]: Sure, I'll fix it

File: src/runtime/SandBox.cpp (hunk 2 above)
Comment (clover2123) [reply src/runtime/SandBox.cpp:170]: You can find stack insertion at line 206 or 224.
FYI) To collect stack tracing information, we traverse state chain and cache state info into `stateStack`

**Review by bbrto21 — APPROVED** [review]

File: src/api/EscargotPublic.cpp (hunk 1 above)
Comment (bbrto21) [reply src/api/EscargotPublic.cpp]: Thank you for hard work!

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**